packages/react/accordion/src/Accordion.tsx

```diff
@@ -86,6 +93,8 @@ const AccordionSingle = React.forwardRef((props, forwardedRef) => {
     value: valueProp,
     defaultValue,
     onValueChange = () => {},
+    collapsible = false,
+    type, // pulling it out as we don't need it and don't want it to end up on the DOM element
```

Comment:
I'm wondering if this should be moved up a level so it wouldn't even exist in these props and then wouldn't need omitting. It's only the parent that needs this prop to determine which one to render.

```jsx
type AccordionOwnProps = 
	| { type: 'single' } & Polymorphic.OwnProps<typeof AccordionSingle>
	| { type: 'multiple' } & Polymorphic.OwnProps<typeof AccordionMultiple>; 

const Accordion = React.forwardRef((props, forwardedRef) => {
	const { type, ...accordionProps } = props;

	if (type === 'single') {
		// ...
	}

	if (type === 'multiple') {
		// ...
	}
});
```

Reply:
I tried that but it didn't let me do it, because of the types.

Reply:
it didn't let you do which part?

Reply:
![image](https://user-images.githubusercontent.com/1539897/118286925-ded7a500-b4ca-11eb-9619-51581eca26a1.png)

Won't work because obviously they are supposed to be parts on it.

Reply:
Right, casting yeah but that's just going around it, seems weird to me, like, we're removing it, but then telling TS it's still there. I'm not sure at this point that it's better than where we're doing it at the moment.

Reply:
That's not weird at all, that's what you have to do with TS when you have union types like this. It doesn't know what the spread props are anymore so you need to hold its hand. That't the way it has always been. 

I'd prefer to not pass the prop down if it's not needed, so the code does what it is supposed to rather than omitting it later with a comment that basically says "we're passing this down unecessarily", just don't pass it no?

Reply:
> It doesn't know what the spread props

That's only half the problem here really, you've removed the `type` so yes, it can't figure out which part of the union it is, you've also removed it, so more importantly because that prop is mandatory it's obviously not in the props anymore so that's really the problem here.

As in, `AccordionSingle` expects `type="single` and `AccordionMultiple` expects `type="multiple"`, it just so happens that we don't use that value, but from a types standpoint, they are required.

Casting fixes both issues at once obviously, but to me I'm thinking the main issue is that we've removed it and now it's missing (from a types standpoint).

Reply:
> we're removing it, but then telling TS it's still there

I'm not telling TS it is still there btw? Maybe there was a misunderstanding somewhere. You would remove it from the Single/Multiple component prop types because they don't need it.

Reply:
I've done it now as I don't think there's any other way to do it in a cleaner fashion.

Reply:
> I'm not telling TS it is still there btw? Maybe there was a misunderstanding somewhere. We would remove it from the Single/Multiple component prop types because they don't need it.

Maybe I'm missing something then…

Reply:
Oh, are you saying we'd do this?

```
type AccordionOwnProps =
  | ({ type: 'single' } & Polymorphic.OwnProps<typeof AccordionSingle>)
  | ({ type: 'multiple' } & Polymorphic.OwnProps<typeof AccordionMultiple>);
```

?

Reply:
Yep, [that's what I said](https://github.com/radix-ui/primitives/pull/651#discussion_r632558843) 😛

Reply:
oh man, I'm freaking blind…